clouddriver-kubernetes-v2/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesCoreCachingAgentTest.java

```diff
@@ -0,0 +1,317 @@
+      getNamedAccountCredentials() {
+    KubernetesV2Credentials v2Credentials1 = mockKubernetesV2Credentials();
+    KubernetesConfigurationProperties.ManagedAccount managedAccount =
+        new KubernetesConfigurationProperties.ManagedAccount();
+    managedAccount.setName(ACCOUNT);
+    managedAccount.setProviderVersion(ProviderVersion.v2);
+    KubernetesV2Credentials.Factory credentialFactory = mock(KubernetesV2Credentials.Factory.class);
+    when(credentialFactory.build(managedAccount)).thenReturn(v2Credentials1);
+    return new KubernetesNamedAccountCredentials<>(managedAccount, credentialFactory);
+  }
+
+  /**
+   * Given a KubernetesNamedAccountCredentials object, builds a set of caching agents responsible
+   * for caching the account's data and returns a collection of those agents.
+   */
+  private static ImmutableCollection<KubernetesCoreCachingAgent> createCachingAgents(
+      KubernetesNamedAccountCredentials<KubernetesV2Credentials> credentials, int agentCount) {
+    ImmutableList.Builder<KubernetesCoreCachingAgent> listBuilder = new ImmutableList.Builder<>();
+    for (int i = 0; i < agentCount; i++) {
+      listBuilder.add(
+          new KubernetesCoreCachingAgent(
+              credentials, objectMapper, new NoopRegistry(), i, agentCount, 10L));
+    }
+    return listBuilder.build();
+  }
+
+  /**
+   * Given an on-demand cache request, constructs a set of caching agents and sends the on-demand
+   * request to those agents, returning a collection of all non-null results of handing those
+   * requests.
+   */
+  private static ImmutableList<OnDemandAgent.OnDemandResult> processOnDemandRequest(
+      Map<String, String> data, ProviderCache providerCache) {
+    KubernetesNamedAccountCredentials<KubernetesV2Credentials> credentials =
+        getNamedAccountCredentials();
+
+    Collection<KubernetesCoreCachingAgent> cachingAgents = createCachingAgents(credentials, 10);
+    ImmutableList.Builder<OnDemandAgent.OnDemandResult> resultBuilder =
+        new ImmutableList.Builder<>();
+    cachingAgents.forEach(
+        cachingAgent -> {
+          OnDemandAgent.OnDemandResult result = cachingAgent.handle(providerCache, data);
+          if (result != null) {
+            resultBuilder.add(result);
+          }
+        });
+    return resultBuilder.build();
+  }
+
+  @Test
+  public void deploymentUpdate() {
+    String expectedKey =
+        String.format(
+            "kubernetes.v2:infrastructure:deployment:%s:%s:%s",
+            ACCOUNT, NAMESPACE1, DEPLOYMENT_NAME);
+
+    ProviderCache providerCache = mock(ProviderCache.class);
+    Map<String, String> data =
+        ImmutableMap.of(
+            "account", ACCOUNT,
+            "location", NAMESPACE1,
+            "name", "deployment " + DEPLOYMENT_NAME);
+    ImmutableList<OnDemandAgent.OnDemandResult> results =
+        processOnDemandRequest(data, providerCache);
+
+    verify(providerCache, times(1)).putCacheData(any(String.class), any(CacheData.class));
```

Comment:
arguably these tests would be better if you used an in-memory cache instead of a mock (`new DefaultProviderCache(new InMemoryCache())`) and tested to see what _actually_ exists in the cache instead of just testing what methods were called...

But I obviously appreciate they likely already took a long time to write and they're certainly better with mocks than not existing at all, so I 100% would understand if you didn't want to rework them all at this point...

In this case, however, I think you'd probably want to specify the arguments to the mocks. `any()` is useful for stubby, but not as great for verification. Can you just move these to the end of the method and do
`verify(providerCache).putCacheData(cacheData.getId(), cacheData)`?

Reply:
Great point...added a commit to use an `InMemoryCache()` instead of stubbing the `DefaultProviderCache`.  I also made a few other improvements to the tests while I was looking at them again.